sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_store_operations.py

```diff
@@ -289,13 +293,15 @@ def begin_update(
                         "any data that was materialized "
                         "earlier will not be available. You have to run backfill again."
                     )
-            elif not materialization_identity:
```

Comment:
why this is removed?

Reply:
We should throw exceptions whenever offline store/online store is provided and materialization identity is not provided or does not exist.

Reply:
Seems the validation logic is moved to line 346.

Reply:
yes, if put it at line 292, this means we will check this materialization identity only when input has offline store and there is an existing offline store, this does not include all the scenarios.